### PR TITLE
Update/add SAGA formulas and provide SAGA installation options in QGIS formulas

### PR DESCRIPTION
**Pull request by pat-s**
### SAGA

Since `saga-gis` (2.2.2) is not working for me (#215), I tried versions 3.0.0 and 4.0.1. Both work and hence I assume that something is broken within 2.2.2 (which also QGIS uses). 

I tried the same example on all versions using `saga_cmd`. 

I also tried 2.2.3 -> also segfault error as for 2.2.2. 

Since I did the work for SAGA v3 and v4 formulas (just changed the download url and sha256), I add two formulas in this PR and renamed `saga-gis` to `saga2` for consistency. 

### QGIS

I also checked SAGA algs in QGIS GUI and got no errors during processing. I do not know if all algorithms are working but I would suggest to have options `--with-saga2`, `--with-saga3` and `--with-saga4` for `qgis2` and `qgis2-ltr` installation. 

I am aware of the fact that SAGA versions >2.2.3 can cause problems in QGIS. So one option would be to add a warning if someone uses the option for SAGA V3 or V4.

### CODE REVIEW

**Comment by dakcarto** [comment]
Hi @pat-s, thanks for contributing (again)!

Try using the new `saga-gis-lts`. It is the only one supported for all current QGIS versions. All `qgis2*` and `qgis3*` formulae have been updated to use it as the `--with-saga-gis` optional dependency.

The current consensus by the QGIS dev team is to not support any versions other than the LTS for Processing, as worked out with the SAGA team, because there were inconsistencies with API wrappers with newer versions.

In light of this, the new options for SAGA 3 and 4 that you have suggested here will not be added to the `qgis` formulae. However, they will make good additions to the base formulae in this tap.

Before they can be added, please download SAGA source code and review what changes have been made to the configure and build setup (I noticed you only changed the download version number). I found several differences between 2.2.2 and 2.3.1, and there will probably be even more between it and the newer versions.

In other words, no new formula can be accepted as mere download version changes for _major_ releases (or, depending upon the project, even minor releases), if it does not accurately represent the build and install docs of the project's specific version. This includes any deprecated or new options added to the project's configuration stage, as well as the default nature of said options.

I understand that this policy is not clearly documented anywhere, but it should always be a consideration when updating large software project's formulae.

**Comment by pat-s** [comment]
> Try using the new saga-gis-lts. It is the only one supported for all current QGIS versions. All qgis2* and qgis3* formulae have been updated to use it as the --with-saga-gis optional dependency.

Works, thanks so much for looking into this and fixing it! 

> The current consensus by the QGIS dev team is to not support any versions other than the LTS for Processing, as worked out with the SAGA team, because there were inconsistencies with API wrappers with newer versions.
> 
> In light of this, the new options for SAGA 3 and 4 that you have suggested here will not be added to the qgis formulae. However, they will make good additions to the base formulae in this tap.

Sure, valid points! 

> Before they can be added, please download SAGA source code and review what changes have been made to the configure and build setup (I noticed you only changed the download version number). I found several differences between 2.2.2 and 2.3.1, and there will probably be even more between it and the newer versions.
> 
> In other words, no new formula can be accepted as mere download version changes for major releases (or, depending upon the project, even minor releases), if it does not accurately represent the build and install docs of the project's specific version. This includes any deprecated or new options added to the project's configuration stage, as well as the default nature of said options.
> 

I understand. I'll see if I have time and motivation to do so. I primarily needed SAGA to work with QGIS since I am an author of the R package `RQGIS` but do not use SAGA actively myself. 

However, we are promoting `osgeo4mac` in our package and I think about writing a blog post about it to make it more popular within the geospatial Mac community (I think it deserves more attention 👍 ). So it would be great if one (or I) could add the missing SAGA 3 and 4 formulas.
